Updated  'Weighing Machine ' exercise. (#1672)
exercism/csharp: exercises/concept/weighing-machine/.docs/instructions.md

> In this exercise you'll be modelling a weighing machine.

Reviewer: Maybe add something about all the weights being in kilograms?

You have 6 tasks each of which requires you to implement one or more properties:

> ## 1. Allow the weighing machine to have a precision.

Reviewer: ```suggestion
## 1. Allow the weighing machine to have a precision
```

> It important to ensure that our weighing machine is precise.

Reviewer: To cater to different demands, we allow each weighing machine to be customized with a precision (the number of digits  after the decimal separator).

Implement the `WeigingMachine` class to have a precision property set only from the constructor:

```csharp
var wm = new WeighingMachine(precision:3);

//  => wm.Precision == 3
```

## 2. Allow the weight to be set on the weighing machine

Implement the `WeigingMachine.Weight` property to allow the weight to be get and set:

```csharp

> var wm = new WeighingMachine(precision:3);

Reviewer: ```suggestion
var wm = new WeighingMachine(precision: 3);
```

> wm.Weight = 60m;

Reviewer: The `m` suffix was needed for it to be a `decimal` type, but we're only using `int` values now. Could you remove the `m` suffix everywhere?

Author: If we use ints then we would also have to remove the precision property which had the goal of students learning about a getter-only property.

//  => wm.Weight == 60m
```

## 3. Ensure that a negative input weight is rejected

> The weight can be set and retrieved, but it can't be negative.

Reviewer: Maybe do something like:

```suggestion
Clearly, someone cannot have a negative weight. 
```

Add validation to the `WeighingMachine.Weight` property to throw an `ArgumentOutOfRangeException` when trying to set it to a negative weight:

```csharp
var wm = new WeighingMachine(precision:3);
wm.Weight = -10m; // Throws an ArgumentOutOfRangeException
```

## 4. Allow a tare adjustment to be applied to the weighing machine

The tare adjustment can be any value (even negative or a value that makes the display weight negative)

> Implement the `WeighingMachine.TareAdjustment` property to allow the tare adjustment to also be set:

Reviewer: ```suggestion
Implement the `WeighingMachine.TareAdjustment` property to allow the tare adjustment to be set:
```

```csharp
var wm = new WeighingMachine();
wm.TareAdjustment = -10;

// => wm.TareAdjustment == -10
```

> ## 5. Ensure that the weighing machine has a default tare Adjustment.

Reviewer: ```suggestion
## 5. Ensure that the weighing machine has a default tare adjustment
```

> A tare adjustment can be applied to the weight they usually have a bias towards overestimating the weight.

Reviewer: ```suggestion
After some thorough testing, it appears that due to a manifacturing issue all weighing machines have a bias towards overestimating the weight by `5`.
```

> Implement the `WeighingMachine.TareAdjustment` property to have the default value as 5.

Reviewer: ```suggestion
Change the `WeighingMachine.TareAdjustment` property to `5` as its default value.
```
